httpclient/api_client.go

```diff
@@ -109,6 +109,7 @@ type DoOption struct {
 	in           RequestVisitor
 	out          func(body *common.ResponseWrapper) error
 	body         any
+	contentType  string
```

Comment:
This duplicates information in the headers. I would just pull `Content-Type` header if set, defaulting to `application/json`.

Reply:
As explained above, the headers have not been set yet (i.e., visitors have not been applied) when marshalling the data.